unittests/conftest.py

```diff
@@ -0,0 +1,47 @@
+from collections.abc import Container, Sequence
+from typing import Iterable, TypeVar
+
+from pvframework.errors import ValidationError
+
+T1 = TypeVar("T1")
+T2 = TypeVar("T2")
+
+
+def intersection_with_contains_str(
+    set_sub_container: Iterable[T1], set_base_container: Iterable[T2]
+) -> tuple[list[T1], list[T2]]:
+    """
+    Returns the intersection of two sets, but only if the intersection is not empty.
+    :param set_sub_container: the set to be checked for intersection
+    :param set_base_container: the set to be checked for intersection
+    :return: the intersection of the two sets, if the intersection is not empty
+    """
+    set_sub_container = list(set_sub_container)
+    set_base_container = list(set_base_container)
+    set_sub_container_intersection = []
+    set_base_container_intersection = []
+    for item_sub in set_sub_container:
+        set_base_remove_list = []
+        for item_base in set_base_container:
+            if str(item_sub) in str(item_base):
+                set_sub_container_intersection.append(item_sub)
+                set_base_container_intersection.append(item_base)
+                set_base_remove_list.append(item_base)
+                break
+        for item_base in set_base_remove_list:
+            set_base_container.remove(item_base)
+
```

Comment:
allein aus dem im docstring beschriebenen zweck versteh ich nicht, warum du nicht die [intersection von `set`](https://docs.python.org/3/library/stdtypes.html#frozenset.intersection) verwendest oder sogar [`isdisjoint`](https://docs.python.org/3/library/stdtypes.html#frozenset.isdisjoint)? (letzteres liest sich ganz ähnlich... hab's aber nicht im detail geprüft). wenn das nen grund hat, kannste ihn ja kommentieren

Reply:
Das Problem in principle: Zwei Mengen `A` und `B`, die jeweils Submengen beinhalten `A_i` und `B_j`. Finde alle `A_i` für die gilt: Es existiert mindestens ein `B_j`, sodass `A_i ⊆ B_j` ist.
Oder mehr am Beispiel: Die tatsächlichen Fehlermeldungen sind superlang. Daher hab ich in die Menge der `expected_errors` nur relevante Auszüge reingeschrieben. Da hilft mir dann die normale intersection leider nicht.

Reply:
Aber ich schreib das nochmal besser in den docstring.

Reply:
https://github.com/Hochfrequenz/pedantic-validator-tool/pull/59/commits/b22c098ea045f872093b61f6c7ae5825e06cee8e